File: scripting/engine/Playground/ecs/components/CameraComponent.swift
@frozen
@component
public struct CameraComponent: Codable {
    public var order: UInt8
    public var fov: Float
    public var nearPlane: Float
    public var farPlane: Float

    public init(
        order: UInt8,
        fov: Float,
        nearPlane: Float,
        farPlane: Float
    ) {
        self.order = order
        self.fov = fov
        self.nearPlane = nearPlane
        self.farPlane = farPlane
    }
}

File: engine/managed/PlaygroundAssembly/ECS/Rendering/CameraComponent.cs
using PlaygroundAssembly.ECS.Core;

namespace PlaygroundAssembly.ECS.Rendering;

[EcsComponent]
public struct CameraComponent(
    byte order,
    float fov,
    float nearPlane,
    float farPlane)
{
    public byte Order = order;
    public float Fov = fov;
    public float NearPlane = nearPlane;
    public float FarPlane = farPlane;
}
